src/lsp/server/diagnostics/DiagnosticProducer.ts

```diff
@@ -8,19 +8,41 @@
 import { Diagnostic } from 'vscode-languageserver/node';
 import { TextDocument } from 'vscode-languageserver-textdocument';
 
-export interface DiagnosticProducer<T> {
+export abstract class DiagnosticProducer<T> {
+    baseDocUrl: string;
```

Comment:
Should it be declared as 'private' since by default, no accessor mean 'public'

Reply:
done